pages/test_analytics/ci_environments.md.erb

```diff
@@ -0,0 +1,108 @@
+# CI Environments
+
+Buildkite Test Analytics collectors automatically detect common continuous integration (CI) environments.
+If available, test collectors gather information about your test runs, such as branch names and build IDs.
+Test collectors gather information from the following CI environments:
+
+- Buildkite
+- CircleCI
+- GitHub Actions
+
+If you run test collectors inside containers or you're using another CI system, then you must set variables to report your CI details to Buildkite.
+
+If you're not using a test collector, see [Importing JSON](/docs/test-analytics/importing-json) and [Importing JUnit XML](/docs/test-analytics/importing-junit-xml) to learn how to provide run environment data.
+
+{:toc}
+
+## Containers and test collectors
+
+If you're using containers within your CI system, then the environment variables used by test collectors may not be exposed to those containers by default.
+Make sure to export your CI environment's variables and your Buildkite API token to your containerized builds and tests.
+
+For example, by default Docker does not receive the host's environment variables.
+To pass them through to the Docker container, use the `--env` option:
+
+```
+  docker run \
+    --env TEST_ANALYTICS_TOKEN \
+    --env BUILDKITE_BUILD_NUMBER \
+    --env BUILDKITE_JOB_ID \
+    --env BUILDKITE_BRANCH \
+    --env BUILDKITE_COMMIT \
+    --env BUILDKITE_MESSAGE \
+    --env BUILDKITE_BUILD_URL \
+    bundle exec rspec
+```
+
+Review the following sections for the environment variables expected by test collectors.
+
```

Comment:
🚨  New section on containerization! 🚨 

Since we can't really know what ridiculous things people are doing with containers (or even what container engine they're using), I've illustrated the concept with plain Docker. Hopefully this gets the point across?

Reply:
:+1: